core/arxiv/submission/rules/size_limits.py

```diff
@@ -0,0 +1,89 @@
+"""Enforces size limit rules."""
+
+from typing import Iterable, Union
+from itertools import count
+import time
+
+from ..domain.event import Event, SetUploadPackage, UpdateUploadPackage, \
+    AddHold, RemoveHold, AddProcessStatus
+from ..domain.event.event import Condition
+from ..domain.submission import Submission, SubmissionContent, Hold
+from ..domain.flag import Flag, ContentFlag
+from ..domain.annotation import Feature
+from ..domain.agent import Agent, User
+from ..domain.process import ProcessStatus
+from ..services import plaintext, compiler
+from ..tasks import is_async
+
+from arxiv.taxonomy import CATEGORIES, Category
+
+PackageEvent = Union[SetUploadPackage, UpdateUploadPackage]
+
+
+COMPRESSED_PACKAGE_MAX = 6_000_000
+UNCOMPRESSED_PACKAGE_MAX = 18_000_000
+PDF_LIMIT = 15_000_000
+"""The maximum size of the resulting PDF."""
+
+
+@SetUploadPackage.bind(lambda *args, **kwargs: True)    # Always check.
+def check_sizes_on_new_source(event: SetUploadPackage, before: Submission,
+                              after: Submission, creator: Agent) \
+        -> Iterable[Event]:
+    """When a new source package is attached, check for oversize source."""
+    return _check_sizes(event, before, after, creator)
+
+
+@UpdateUploadPackage.bind(lambda *args, **kwargs: True)    # Always check.
+def check_sizes_on_update_source(event: UpdateUploadPackage,
+                                 before: Submission, after: Submission,
+                                 creator: Agent) -> Iterable[Event]:
+    """When a source package is updated, check for oversize source."""
+    return _check_sizes(event, before, after, creator)
+
+
+def pdf_is_compiled(event: Event, *args, **kwargs) -> bool:
```

Comment:
Should this method be bound to a particular event?

Reply:
This is use as the condition for the callback, on L51. This way we are only running ``check_pfdf_size`` when the process being updated is for compilation, and the status is ``SUCCEEDED``.